addons/sourcemod/scripting/Leader.sp

```diff
@@ -1125,150 +1628,811 @@ public void MarkerMenu(int client)
 	else
 	beacon = "No";
 
-	SetMenuTitle(menu, "Leader Menu\nSprite: %s\nMarker: %s\nBeacon: %s", sprite, marker, beacon);
-	AddMenuItem(menu, "removemarker", "Remove Marker");
-	AddMenuItem(menu, "defendmarker", "Defend Marker");
+	char sTitle[200];
+	Format(sTitle, sizeof(sTitle), "Leader Menu\nSprite: %s\nMarker: %s\nBeacon: %s", sprite, marker, beacon);
+	menu.SetTitle(sTitle);
+	menu.AddItem("removemarker", "Remove Marker");
+	menu.AddItem("defendmarker", "Defend Marker");
+	menu.AddItem("signal", "Signal Marker with Aim");
 
-	SetMenuExitBackButton(menu, true);
-	SetMenuExitButton(menu, true);
-	DisplayMenu(menu, client, MENU_TIME_FOREVER);
+	menu.ExitBackButton = true;
+	menu.Display(client, MENU_TIME_FOREVER);
 }
 
 //----------------------------------------------------------------------------------------------------
 // Purpose:
 //----------------------------------------------------------------------------------------------------
-public int MarkerMenu_Handler(Handle menu, MenuAction action, int client, int position)
+public int MarkerMenu_Handler(Menu menu, MenuAction action, int client, int position)
 {
-	if(leaderClient == client && IsValidClient(client))
+	if(IsValidClient(client) && g_bIsClientALeader[client])
 	{
 		if(action == MenuAction_Select)
 		{
-			char info[32];
-			GetMenuItem(menu, position, info, sizeof(info));
-
-			if(StrEqual(info, "removemarker"))
+			switch(position)
 			{
-				RemoveMarker(client);
-				CPrintToChat(client, "{green}[SM] {default}Marker removed.");
-				markerActive = false;
-				LeaderMenu(client);
+				case 0:
+				{
+					RemoveMarker(client);
+					CPrintToChat(client, "{green}[SM] {default}Marker removed.");
+					markerActive[client] = false;
+					LeaderMenu(client);
+				}
+				case 1:
+				{
+					RemoveMarker(client);
+					markerEntities[client] = SpawnMarker(client, DefendVMT);
+					CPrintToChat(client, "{green}[SM] {default}Marker {green}Defend Here {default}placed.");
+					markerActive[client] = true;
+					LeaderMenu(client);
+				}
+				case 2:
+				{
+					SignalMenu(client);
+				}
 			}
-			if(StrEqual(info, "defendmarker"))
+		}
+		else if(action == MenuAction_End)
+			delete menu;
+			
+		else if (action == MenuAction_Cancel && position == MenuCancel_ExitBack)
+			LeaderMenu(client);
+	}
+	return 0;
+}
+
+//----------------------------------------------------------------------------------------------------
+// Purpose:
+//----------------------------------------------------------------------------------------------------
+public void SignalMenu(int client)
+{
+	Menu menu = new Menu(SignalMenu_Handler);
+
+	char color[64], sButton[64];
+	switch(g_iClientColorNum[client])
+	{
+		case 0:
+			color = "White";
+		case 1:
+			color = "Red";
+		case 2:
+			color = "Lime";
+		case 3:
+			color = "Blue";
+		case 4:
+			color = "Yellow";
+		case 5:
+			color = "Cyan";
+		case 6:
+			color = "Gold";
+		case 7:
+			color = "Random";
+	}
+	
+	switch(g_iClientButton[client])
```

Comment:
put this in a function like `stock void GetClientButtonName(int client, char[] buffer, int bufferSize)`